task_processing/plugins/mesos/execution_framework.py

```diff
@@ -55,17 +57,28 @@ def __init__(
             checkpoint=True,
             role=self.role
         )
-        self.task_queue = Queue(max_task_queue_size)
-        self.task_update_queue = Queue(max_task_queue_size)
+
+        self.task_queue = make_thread_safe_struct(
```

Comment:
Hm, I'm concerned about this. Afaik Python [Queues](https://docs.python.org/3/library/queue.html#module-queue) _are_ thread safe (as are [collections.dequeue](https://docs.python.org/3/library/collections.html#collections.deque)). We don't need to add locking on top of them IMO unless we have a very specific re-entrant case. If we need multi-process support multiprocessing has a Queue equivalent which is quite good too.

I'm a much bigger fan of using pyrsistent datastructures everywhere and for the pubsub using well established stdlib queues (or a package). I don't think we want to be implementing our own locked datastructures.